nox/_options.py

```diff
@@ -12,23 +12,202 @@
+    description="Nox is a Python automation toolkit.", add_help=False
+)
+
+options.add_group(
+    "primary",
+    "Primary arguments",
+    "These are the most common arguments used when invoking Nox.",
+)
+options.add_group(
+    "secondary",
+    "Additional arguments & flags",
+    "These arguments are used to control Nox's behavior or control advanced features.",
+)
+
+
+def _sessions_and_keywords_merge_func(key, command_args, noxfile_args):
+    """Only return the Noxfile value for sessions/keywords if neither sessions
+    or keywords are specified on the command-line."""
+    if not command_args.sessions and not command_args.keywords:
+        return getattr(noxfile_args, key)
+    return getattr(command_args, key)
+
+
+def _envdir_merge_func(command_args, noxfile_args):
+    """Ensure that there is always some envdir."""
+    return command_args.envdir or noxfile_args.envdir or ".nox"
+
+
+def _sessions_default():
+    """Looks at the NOXSESSION env var to set the default value for sessions."""
+    nox_env = os.environ.get("NOXSESSION")
+    env_sessions = nox_env.split(",") if nox_env else None
+    return env_sessions
+
+
+def _color_finalizer(value, args):
+    """Figures out the correct value for "color" based on the two color flags."""
+    return not args.nocolor or args.forcecolor
+
+
+def _posargs_finalizer(value, args):
+    """Removes any leading "--"s in the posargs array."""
+    if value and value[0] == "--":
+        value.pop(0)
+    return value
+
+
+options.add_options(
+    _option_set.Option(
+        "help",
+        "-h",
+        "--help",
+        group="primary",
+        action="store_true",
+        help="Show this help message and exit.",
+    ),
+    _option_set.Option(
+        "version",
+        "--version",
+        group="primary",
+        action="store_true",
+        help="Show the Nox version and exit.",
+    ),
+    _option_set.Option(
+        "list_sessions",
+        "-l",
+        "--list-sessions",
+        "--list",
+        group="primary",
+        action="store_true",
+        help="List all available sessions and exit.",
+    ),
+    _option_set.Option(
+        "sessions",
+        "-s",
+        "-e",
+        "--sessions",
+        group="primary",
+        noxfile=True,
+        merge_func=functools.partial(_sessions_and_keywords_merge_func, "sessions"),
+        nargs="*",
+        default=_sessions_default,
+        help="Which sessions to run, by default, all sessions will run.",
+    ),
+    _option_set.Option(
+        "keywords",
+        "-k",
+        "--keywords",
+        noxfile=True,
+        merge_func=functools.partial(_sessions_and_keywords_merge_func, "keywords"),
+        help="Only run sessions that match the given expression.",
+    ),
+    _option_set.Option(
+        "posargs",
+        "posargs",
+        group="primary",
+        nargs=argparse.REMAINDER,
+        help="Arguments following -- that are passed through to the session(s).",
```

Comment:
Here the `--` just looks like part of the sentence (like an `endash`). Maybe put quotes or backticks around it?

Reply:
Done.